server/index.js

```diff
@@ -85,10 +85,38 @@ app.get("/api/eggs", (req, res, next) => {
   return db
     .query(sql)
     .then((result) => {
-      const [egg] = result.rows;
-      egg.longitude = Number(egg.longitude);
-      egg.latitude = Number(egg.latitude);
-      res.status(200).json([egg]);
+      const egg = result.rows;
+      egg.forEach((egg) => {
+        egg.latitude = Number(egg.latitude);
+        egg.longitude = Number(egg.longitude);
+      });
+      res.status(200).json(egg);
+    })
+    .catch((err) => next(err));
+});
+
+app.get("/api/details", (req, res, next) => {
+  const eggId = Number(req.headers.eggid);
+
+  console.log(eggId);
+  if (!eggId) throw new ClientError(400, "invalid request for egg");
```

Comment:
Assuming that your egg ids are integers, you would check to make sure it's an integer and then let the client know that the param needs to be an integer.